### ISSUE TITLE
Missing files/folders with Japanese character is its name after pack.

### ISSUE THREAD

**Issue by Andrei-Stakov** [AtomCrafty/yukatool2#4]
This might be related to AtomCrafty/yukatool#8

`Unpack` / `decompile `was working spotlessly (all Japanese-character filename is safely extracted), but I hit a roadblock while using `pack`. Note that the `pack` itself is a success with no warning present.

The repacked .ykc is seemingly fine, size-wise (within a bit of margin of error). But the game having missing file popups; it shows when the missing files have Japanese character in its filename.

After checking it with YukaTool GUI, I found a common issue; few file and folders were completely missing.
![image](https://user-images.githubusercontent.com/16612525/108367523-23390900-722c-11eb-9545-295bfa1955d0.png)

Browsing around the archive, I found that every single files with Japanese-character in its filename, with any folder that only contain said files (no other, plain ASCII-filenamed files present) were missing.

**Comment by AtomCrafty** [comment]
Hm, this is weird... Could you post one of the Japanese file names to I can try to reproduce the issue?

**Comment by Andrei-Stakov** [comment]
Here's a few of them, within relative path:

`.\bg\【随時更新】背景リソース欠けてるデータ.bak` — [Since `bg` folder also have files using plain ASCII filename, it still showed up. Related files still missing though]

`.\face\あやめａｂ_制服_フェイス_１.png` — [`face `folder was missing because the whole files inside were using Japanese character]

`.\ta\ayame\あやめａ_制服_ｓｓ_６.png` — [Same reason and situation as the 2nd sample]

Also, here's a sample directory case, just in case you need it:
[data_test.zip](https://github.com/AtomCrafty/yukatool2/files/6004498/data_test.zip)
Sorry for the big size.

**Comment by AtomCrafty** [comment]
Thanks, that sample data helped me pinpoint the issue!

So it turns out the files were all correctly written to the archive and even the names were encoded correctly, but in the index (basically the archive's table of contents) the file name length was being calculated incorrectly. So all the data was in there, but couldn't be accessed because of the invalid index entries.

I've created a [new release](https://github.com/AtomCrafty/yukatool2/releases/tag/v2.4) that fixes the issue, please check if it works for you :)

**Comment by Andrei-Stakov** [comment]
It's a fine success 👍 !

![image](https://user-images.githubusercontent.com/16612525/108441112-07af1c00-7287-11eb-9b5a-17ab87f7feec.png)

The funny thing is, the game's engine is kinda bonker. The same popups still shown, even though the files are there.

Then I noticed one thing; the related path for the files shown in the popup were using _uppercase_, including the folders. Might go broke as well to set those folder and files that way.

_**And it runs.**_
Even though it was essentially just a repackaged .ykc, it needs an uppercase conversion since this game's engine version seemingly only accept case-sensitive filename, at least for the related files.

This might be a combination of few possible points:
- The engine itself is only recognizing case-sensitive filenames (_ARGH!_).
- YukaTool somehow ignored or can't recognized the filename case and just use lowercase for decompiled files on `unpack`.
- YukaTool properly preserved the filename case on `pack`.

I'll create a new issue for this as usual, so I'll close the current issue since ti was solved perfectly. Thanks for the awesome help!
